---
title: "About"
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

## Who

*  [PoYuan Tung][tung] is a postdoctoral fellow in the Department of Human Genetics at the University of Chicago.

*  [John Blischak][blischak] is a graduate student in the Committee on Genetics, Genomics, and Systems Biology at the University of Chicago.

*  [Joyce Hsiao][hsiao] is a postdoctoral fellow in the Department of Human Genetics at the University of Chicago.

*  [David Knowles][knowles] is a postdoctoral fellow at Stanford University.

*  [Jonathan Pritchard][pritchard] is a Professor in the Departments of Genetics and Biology at Stanford University and an Investigator in the Howard Hughes Medical Institute.

*  [Yoav Gilad][gilad] is a Professor in the Department of Human Genetics at the University of Chicago.

[tung]: https://github.com/pytung
[blischak]: https://github.com/jdblischak
[hsiao]: https://github.com/jhsiao999
[knowles]: https://github.com/davidaknowles
[pritchard]: http://pritchardlab.stanford.edu/pritchard.html
[gilad]: http://genes.uchicago.edu/directory/yoav-gilad-phd/

## What

We study the role of interindividual genetic variation on gene regulatory phenotypes.
In this context, we are interested in single cell gene expression data because we would like to characterize how genetic variation impacts the variability in cell-cell gene expression between individuals.
In other words, whereas traditional eQTL mapping identifies associations between SNPs and mean gene expression levels, we'd like to idenitfy QTLs that are associated with the variance in gene expression levels.
For this project, we are exploring multiple questions, including:

*  How many cells do we need to collect and at what depth do we need to sequence each cell to design effective single cell studies?
*  How can we identify and mitigate the technical noise introduced in the batch processing of single cell data?
*  What measure of variance should we use to test for differences in cell-cell variability?
*  Which genes are differentially variable between individuals?

To view our results, read our paper [Batch effects and the effective design of single-cell gene expression studies](http://www.nature.com/articles/srep39921).

To learn more about our other research projects, visit our [lab website](http://giladlab.uchicago.edu).

## Why

We have created this website because we wanted to experiment with performing science in the open.
From the beginning of the project all of our results have been posted to this site.
By doing so, we hope that our results are more transparent and reproducible, and also more quickly communicated to the rest of the scientific community.

## How

We made this website using many wonderful publicly available tools.
We write the analyses using [R Markdown][rmarkdown], [knitr][], and [pandoc][].
In fact, the initial template for this site was modeled after the [R Markdown website][rmd-site].
To host the site, we use [GitHub Pages][gh-pages].

**Note:** This site was created before the [rmarkdown][] package had helper functions for creating websites, and thus is more complicated than it would need to be if we were creating it today.
If you're interested in creating a similar site, start by reading the documentation on [R Markdown Websites][web].

[rmarkdown]: http://rmarkdown.rstudio.com/
[knitr]: http://yihui.name/knitr/
[pandoc]: http://pandoc.org/
[rmd-site]: https://github.com/rstudio/rmarkdown/tree/gh-pages
[gh-pages]: https://pages.github.com/
[web]: http://rmarkdown.rstudio.com/rmarkdown_websites.html
