---
title: "Data transformation"
date: 2015-09-30
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
opts_chunk$set(fig.width = 8, fig.height = 8)
```

```{r packages, message=FALSE}
library("ggplot2")
library("cowplot")
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank(),
             legend.key = element_blank(),
             plot.title = element_text(size = rel(1)))
source("functions.R")
```

This file performs principal components analysis (PCA) and displays the results for the data at each stage of our data transformation pipeline.
Furthermore, it quantifies and tests the PCA results using [pca-utils][] (commit [8540ff2][]).
The methodology is described in [Worley et al., 2013][worley2013].
Briefly, the differences between samples are calculated using the squared Mahalanobis distance.
The distances are then scaled so that the hypothesis test that the samples from two groups come from separate distributions can be computed with an F-test.

[pca-utils]: https://github.com/geekysuavo/pca-utils
[8540ff2]: https://github.com/geekysuavo/pca-utils/tree/8540ff2c6769031b0505af78f2dcb5703ae09294
[worley2013]: http://www.ncbi.nlm.nih.gov/pubmed/23079505

## Input

Input filtered annotation.

```{r input-annotation-filter}
anno_filter <- read.table("../data/annotation-filter.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno_filter)
```

Input filtered molecule counts.

```{r input-molecule-counts-filter}
molecules_filter <- read.table("../data/molecules-filter.txt", header = TRUE,
                               stringsAsFactors = FALSE)
stopifnot(ncol(molecules_filter) == nrow(anno_filter),
          colnames(molecules_filter) == anno_filter$sample_id)
```

Input standardized molecule counts.

```{r input-molecule-counts-cpm}
molecules_cpm <- read.table("../data/molecules-cpm.txt", header = TRUE,
                            stringsAsFactors = FALSE)
stopifnot(ncol(molecules_cpm) == nrow(anno_filter),
          colnames(molecules_cpm) == anno_filter$sample_id)
```

Input Poisson GLM transformed molecule counts per million.

```{r input-molecule-counts-trans}
molecules_cpm_trans <- read.table("../data/molecules-cpm-trans.txt", header = TRUE,
                               stringsAsFactors = FALSE)
stopifnot(ncol(molecules_cpm_trans) == nrow(anno_filter),
          colnames(molecules_cpm_trans) == anno_filter$sample_id)
```

Input final batch-corrected molecule counts per million.

```{r input-molecule-counts-final}
molecules_final <- read.table("../data/molecules-final.txt", header = TRUE,
                              stringsAsFactors = FALSE)
stopifnot(ncol(molecules_final) == nrow(anno_filter),
          colnames(molecules_final) == anno_filter$sample_id)
```

## PCA

### PCA of filtered data

```{r pca-molecules-filter}
pca_molecules_filter <- run_pca(molecules_filter)
pca_molecules_filter_title <- "The effect of technical batch on single cell gene expression for raw counts data"
pca_molecules_filter_plot <- plot_pca(pca_molecules_filter$PCs,
         explained = pca_molecules_filter$explained,
         metadata = anno_filter, color = "individual",
         shape = "replicate", alpha = 0.5, size = 2.2) +
  # labs(title = "The effect of technical batch \n on single cell gene expression data \n when raw counts are used")
  labs(title = paste(strwrap(pca_molecules_filter_title, width = 50), collapse = "\n"))
```

```{r pca-molecules-filter-out}
pca_molecules_filter_out <- data.frame(Obs.id = 1:nrow(anno_filter),
                                       Obs.batch = anno_filter$batch,
                                       pca_molecules_filter$PCs[, 1:2])
write.table(pca_molecules_filter_out,
            file = "../data/pca-molecules-filter.txt",
            quote = FALSE, sep = "\t", row.names = FALSE)
```

```{r pca-molecules-filter-dist, engine='bash'}
pca-distances -i ../data/pca-molecules-filter.txt
```

```{r pca-molecules-filter-dendro, engine='bash'}
pca-dendrogram -i ../data/pca-molecules-filter.txt
```

### PCA of standardized data

```{r pca-molecules-cpm}
pca_molecules_cpm <- run_pca(molecules_cpm)
pca_molecules_cpm_title <- "The effect of technical batch on single cell gene expression for log counts per million data"
pca_molecules_cpm_plot <- plot_pca(pca_molecules_cpm$PCs,
         explained = pca_molecules_cpm$explained,
         metadata = anno_filter, color = "individual",
         shape = "replicate", alpha = 0.5, size = 2.2) +
  labs(title = paste(strwrap(pca_molecules_cpm_title, width = 50), collapse = "\n"))
  # labs(title = "The effect of technical batch \n on single cell gene expression data \n when log transformed counts per million data are used")
```

```{r pca-molecules-cpm-out}
pca_molecules_cpm_out <- data.frame(Obs.id = 1:nrow(anno_filter),
                                       Obs.batch = anno_filter$batch,
                                       pca_molecules_cpm$PCs[, 1:2])
write.table(pca_molecules_cpm_out,
            file = "../data/pca-molecules-cpm.txt",
            quote = FALSE, sep = "\t", row.names = FALSE)
```

```{r pca-molecules-cpm-dist, engine='bash'}
pca-distances -i ../data/pca-molecules-cpm.txt
```

```{r pca-molecules-cpm-dendro, engine='bash'}
pca-dendrogram -i ../data/pca-molecules-cpm.txt
```

### PCA of Poisson GLM transformed molecule counts per million

```{r pca-molecules-cpm-trans}
pca_molecules_cpm_trans <- run_pca(molecules_cpm_trans)
pca_molecules_cpm_trans_title <- "The effect of technical batch on single cell gene expression for Poisson transformed data"
pca_molecules_cpm_trans_plot <- plot_pca(pca_molecules_cpm_trans$PCs,
         explained = pca_molecules_cpm_trans$explained,
         metadata = anno_filter, color = "individual",
         shape = "replicate",  alpha = 0.5, size = 2.2) +
  labs(title = paste(strwrap(pca_molecules_cpm_trans_title, width = 50), collapse = "\n"))
  # labs(title = "The effect of technical batch \n on single cell gene expression data \n when Poisson transformed data are used")
```

```{r pca-molecules-cpm-trans-out}
pca_molecules_cpm_trans_out <- data.frame(Obs.id = 1:nrow(anno_filter),
                                       Obs.batch = anno_filter$batch,
                                       pca_molecules_cpm_trans$PCs[, 1:2])
write.table(pca_molecules_cpm_trans_out,
            file = "../data/pca-molecules-cpm-trans.txt",
            quote = FALSE, sep = "\t", row.names = FALSE)
```

```{r pca-molecules-cpm-trans-dist, engine='bash'}
pca-distances -i ../data/pca-molecules-cpm-trans.txt
```

```{r pca-molecules-cpm-trans-dendro, engine='bash'}
pca-dendrogram -i ../data/pca-molecules-cpm-trans.txt
```

### PCA of final batch-corrected data

```{r pca-molecules-final}
pca_final <- run_pca(molecules_final)
pca_final_title <- "PCA of standardized, normalized, and batch corrected single cell gene expression data"
pca_final_plot <- plot_pca(pca_final$PCs, explained = pca_final$explained,
         metadata = anno_filter, color = "individual",
         shape = "replicate", alpha = 0.5, size = 2.2) +
  labs(title = paste(strwrap(pca_final_title, width = 50), collapse = "\n"))
  # labs(title = "PCA of standardized, \n normalized, and batch corrected \n single cell gene expression data")
```

```{r pca-final-out}
pca_final_out <- data.frame(Obs.id = 1:nrow(anno_filter),
                            Obs.batch = anno_filter$batch,
                            pca_final$PCs[, 1:2])
write.table(pca_final_out,
            file = "../data/pca-final.txt",
            quote = FALSE, sep = "\t", row.names = FALSE)
```

```{r pca-molecules-final-dist, engine='bash'}
pca-distances -i ../data/pca-final.txt
```

```{r pca-molecules-final-dendro, engine='bash'}
pca-dendrogram -i ../data/pca-final.txt
```

## Data transformation figure

```{r data-transformation, fig.width=12, fig.height=12}
plot_final <- plot_grid(
  pca_molecules_filter_plot +
    scale_color_discrete(name = "Individual") +
    guides(shape = FALSE) + theme(legend.position = "bottom"),
  pca_molecules_cpm_plot +
    scale_shape(name = "Replicate") +
    guides(color = FALSE) + theme(legend.position = "bottom"),
  pca_molecules_cpm_trans_plot +
    scale_color_discrete(name = "Individual") +
    guides(shape = FALSE) + theme(legend.position = "bottom"),
  pca_final_plot +
    scale_shape(name = "Replicate") +
    guides(color = FALSE) + theme(legend.position = "bottom"),
  labels = letters[1:4])
plot_final
tiff("../paper/figure/fig-data-transformation.tiff",
     width = 12, height = 12,
     units = "in", res = 300, compression = "none")
plot_final
dev.off()
tiff("../paper/figure/fig-data-transformation.tiff",
     width = 12, height = 12,
     units = "in", res = 300, compression = "zip")
plot_final
dev.off()
# An uncompressed version to put in Word
tiff("../paper/figure/fig-data-transformation-uncompressed.tiff",
     width = 12, height = 12,
     units = "in", res = 300, compression = "none")
plot_final
dev.off()
```

## Session information

```{r info}
sessionInfo()
```
