---
title: "Quality control plots"
author: "PoYuan Tung"
date: 2015-10-21
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
opts_chunk$set(cache = FALSE)
```

## Input

```{r packages, message=FALSE, cache=FALSE}
library("biomaRt")
library("data.table")
library("dplyr")
library("limma")
library("edgeR")
library("ggplot2")
library("grid")
library("cowplot")
theme_set(theme_bw(base_size = 16))
theme_update(panel.grid.minor.x = element_blank(),
             panel.grid.minor.y = element_blank(),
             panel.grid.major.x = element_blank(),
             panel.grid.major.y = element_blank(),
             legend.key = element_blank(),
             plot.title = element_text(size = rel(1)))
source("functions.R")
```

Input ERCC molecule counts calculated in [capture efficiency](capture-efficiency.html). 

```{r input-ercc}
ercc <- read.table("../data/expected-ercc-molecules.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(ercc)
```

Input annotation.

```{r input-annotation}
anno <- read.table("../data/annotation.txt", header = TRUE,
                   stringsAsFactors = FALSE)
head(anno)
```

Input read counts and filter for quality cells.

```{r input-read-counts}
reads <- read.table("../data/reads.txt", header = TRUE,
                    stringsAsFactors = FALSE)

quality_single_cells <- scan("../data/quality-single-cells.txt", what = "character")
reads <- reads[, colnames(reads) %in% quality_single_cells]
```

Input read counts in high quality cells for filtered genes

```{r input-read-filter-counts}
reads_filter <- read.table("../data/reads-filter.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

Input molecule counts and filter for quality cell.

```{r input-molecule-counts}
molecules <- read.table("../data/molecules.txt", header = TRUE,
                    stringsAsFactors = FALSE)

molecules <- molecules[, colnames(molecules) %in% quality_single_cells]
```

Input molecule counts in high quality cells for filtered genes

```{r input-molecule-filter-counts}
molecules_filter <- read.table("../data/molecules-filter.txt", header = TRUE,
                    stringsAsFactors = FALSE)
```

## Compare reads and molecules

Compare the means of each gene obtained via the different methods.

```{r reads-v-molecules}
## calculate mean
reads_mean     <- apply(reads, 1, mean)
molecules_mean <- apply(molecules, 1, mean)
distribution <- data.frame(reads_mean, molecules_mean)

reads_filter_mean <- apply(reads_filter, 1, mean)
molecules_filter_mean <- apply(molecules_filter, 1, mean)
distribution_filter <- data.frame(reads_filter_mean, molecules_filter_mean)

## correlation between reads and molecules
cor(distribution)
cor(distribution_filter)

## select ERCC
distribution$type <- ifelse(grepl("ERCC", rownames(distribution)), "ERCC", "gene")
distribution_filter$type <- ifelse(grepl("ERCC", rownames(distribution_filter)), "ERCC", "gene")

## color palette
cbPalette <- c("#0000FF", "#999999", "#990033", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "#009E73")

## plots
plot_mean_log <- ggplot(distribution, aes(x = molecules_mean, y = reads_mean, col = type)) + 
  geom_point(size = 3, alpha = 0.5) +  
  scale_colour_manual(values=cbPalette) + 
  labs(x = "Mean molecule-counts across samples", 
       y = "Mean read-counts across samples", 
       title = "Amplification bias is stronger \n in endogenous genes than in ERCC spike-ins") + 
  scale_x_log10() + 
  scale_y_log10()

plot_mean_log

plot_mean_filter_log <- ggplot(distribution_filter, aes(x = molecules_filter_mean, y = reads_filter_mean, col = type)) + 
  geom_point(size = 3, alpha = 0.5) +  
  scale_colour_manual(values=cbPalette) + 
  labs(x = "Mean molecule-counts across samples", 
       y = "Mean read-counts", 
       title = "Read-counts and molecule-counts of each gene (log scale)") + 
  scale_x_log10() + 
  scale_y_log10()

plot_mean_filter_log

plot_mean <- ggplot(distribution, aes(x = molecules_mean, y = reads_mean, col = type)) + 
  geom_point(size = 3, alpha = 0.5) +  
  scale_colour_manual(values=cbPalette) + 
  labs(x = "Mean molecule-counts across samples", 
       y = "Mean read-counts across samples", 
       title = "Amplification bias is stronger \n in endogenous genes than in ERCC spike-ins") +
  geom_smooth(method = "lm")

plot_mean

```

## Distribution of fold change to mean

Look at the distribution of fold change to mean. As being reported by others, the lowly expressed genes show divergent read and molecule counts

```{r distribution}
## calculate fold change to mean
distribution$fold_change_read     <- log2(reads_mean/mean(reads_mean))
distribution$fold_change_molecule <- log2(molecules_mean/mean(molecules_mean))

plot_distribution <- ggplot(distribution, aes(x = fold_change_molecule, y = fold_change_read, col = type)) + 
  geom_point(size = 3, alpha = 0.5) + 
  scale_colour_manual(values=cbPalette) + 
  stat_function(fun= function(x) {x}, col= "#56B4E9") + 
  labs(x = "log2 fold change to mean (molecule)", y =  "log2 fold change to mean (reads)")

plot_distribution
```

## Visualizing capture efficiency

Use only those `r sum(ercc$ercc_molecules_well >= 1)` ERCC genes with at least 1 expected molecule per well.

```{r capture-efficiency}
ercc_list <- list()
for (spike in ercc$id[ercc$ercc_molecules_well >= 1]) {
  if (spike %in% rownames(molecules)) {
    ercc_list$id <- c(ercc_list$id, spike)
    ercc_list$observed_mean <- c(ercc_list$observed_mean,
                                 mean(as.numeric(molecules[spike, ])))
    ercc_list$observed_sem <- c(ercc_list$observed_sem,
                                 sd(as.numeric(molecules[spike, ])) /
                                 sqrt(ncol(molecules)))
    ercc_list$expected <- c(ercc_list$expected,
                            ercc$ercc_molecules_well[ercc$id == spike])
  }
}
ercc_plot <- as.data.frame(ercc_list, stringsAsFactors = FALSE)
str(ercc_plot)
cor(ercc_plot$observed_mean, ercc_plot$expected)
```

Use molecule filter file. 

```{r capture-efficiency-filter}
ercc_list_filter <- list()
for (spike in ercc$id[ercc$ercc_molecules_well >= 0]) {
  if (spike %in% rownames(molecules_filter)) {
    ercc_list_filter$id <- c(ercc_list_filter$id, spike)
    ercc_list_filter$observed_mean <- c(ercc_list_filter$observed_mean,
                                 mean(as.numeric(molecules_filter[spike, ])))
    ercc_list_filter$observed_sem <- c(ercc_list_filter$observed_sem,
                                 sd(as.numeric(molecules_filter[spike, ])) /
                                 sqrt(ncol(molecules_filter)))
    ercc_list_filter$expected <- c(ercc_list_filter$expected,
                            ercc$ercc_molecules_well[ercc$id == spike])
  }
}
ercc_filter_plot <- as.data.frame(ercc_list_filter, stringsAsFactors = FALSE)
str(ercc_filter_plot)
cor(ercc_filter_plot$observed_mean, ercc_filter_plot$expected)
```

```{r ercc-capture-efficiency, fig.width=8, fig.height=8}
p_efficiency <- ggplot(ercc_plot, aes(x = expected, y = observed_mean, label = id)) +
  geom_point(col = "red") +
  geom_errorbar(aes(ymin = observed_mean - observed_sem,
                    ymax = observed_mean + observed_sem), width = 0) +
  labs(x = "Expected ERCC molecules",
       y = "Observed ERCC molecules\n(mean +/- SEM)",
       title = "ERCC capture efficiency")

p_efficiency_plot <- p_efficiency + scale_x_log10() +
  scale_y_log10() +
  labs(x = "Expected ERCC molecules",
       y = "Observed ERCC molecules\n(mean +/- SEM)",
       title = "ERCC capture efficiency \n (48 ERCC spike-in controls)")

p_efficiency_plot + geom_text(hjust = 0, nudge_x = 0.05, size = 4)

p_efficiency_filter_plot <- ggplot(ercc_filter_plot, aes(x = expected, y = observed_mean)) +
  geom_point(col = "red") +
  geom_errorbar(aes(ymin = observed_mean - observed_sem,
                    ymax = observed_mean + observed_sem), width = 0) +
                    scale_x_log10() + scale_y_log10() +
                    labs(x = "Expected ERCC molecules",
                    y = "Observed ERCC molecules\n(mean +/- SEM)",
                    title = "ERCC capture efficiency")

p_efficiency_filter_plot
```

## Calculate capture efficiency per cell
```{r capture-efficiency-cell}
ercc_index <- grep("ERCC", rownames(molecules_filter))
length(ercc_index)

efficiency <- numeric(length = ncol(molecules_filter))
total_ercc_molecules <- sum(ercc_filter_plot$expected)
for (i in 1:ncol(molecules_filter)) {
  efficiency[i] <- sum(molecules_filter[ercc_index, i]) / total_ercc_molecules
}
summary(efficiency)
```

## QC plots for paper
```{r plots, fig.width=12, fig.height=4, dpi=300}
plot_grid(plot_mean_log + theme(legend.position = c(.85,.25)) + labs (col = ""),
          p_efficiency_plot + theme(legend.position = "none"),
          labels = letters[6:7])
```

## Session information

```{r info}
sessionInfo()
```
