---
title: Legends
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Figure Legends

**Figure `r fig_main_qc`. Experimental design and quality control of scRNA-seq.**
\(A\) Three C1 96 well-integrated fluidic circuit (IFC) replicates were collected from each of the three Yoruba individuals.
A bulk sample was included in each batch.
(B) Summary of the cutoffs used to remove data from low quality cells that might be ruptured or dead (See Supplementary Fig. `r fig_supp_qc` for details).
(C-E) To assess the quality of the scRNA-seq data, the capture efficiency of cells and the faithfulness of mRNA fraction amplification were determined based on the proportion of unmapped reads, the number of detected genes, the numbers of total mapped reads, and the proportion of ERCC spike-in reads across cells.
The dash lines indicate the cutoffs summarized in panel (B).
The three colors represent the three individuals (NA19098 in red, NA19101 in green, and NA19239 in blue), and the numbers indicate the cell numbers observed in each capture site on C1 plate.
(F) Scatterplots in log scale showing the mean read counts and the mean molecule counts of each endogenous gene (grey) and ERCC spike-ins (blue) from the `r nrow(anno_filter)` high quality single cell samples before removal of genes with low expression.
(G) mRNA capture efficiency shown as observed molecule count versus number of molecules added to each sample, only including the `r sum(grepl("ERCC", rownames(molecules_filter)))` ERCC spike-in controls remaining after removal of genes with low abundance.
Each red dot represents the mean +/- SEM of an ERCC spike-in across the `r nrow(anno_filter)` high quality single cell samples.

**Figure `r fig_main_subsample`. The effect of sequencing depth and cell number on single cell UMI estimates.**
Sequencing reads from all the high quality single cells collected for NA19239 were subsampled to the indicated sequencing depth and cell number, and subsequently converted to molecules using the UMIs. Each point represents the mean +/- SEM of 10 random draws of the indicated cell number.
The left panel displays the results for 6,097 (50% of detected) genes with lower expression levels  and the right panel the results for 6,097 genes with higher expression levels.
\(A\) Pearson correlation of aggregated gene expression level estimates from single cells compared to the bulk sequencing samples.
(B) Total number of genes detected with at least one molecule in at least one of the single cells.
(C) Pearson correlation of cell-to-cell gene expression variance estimates from subsets of single cells compared to the full single cell data set.

**Figure `r fig_main_batch`. Batch effect of scRNA-seq data using the C1 platform.**
\(A\) Violin plots of the number of total ERCC spike-in molecule-counts in single cell samples per C1 replicate.
(B) Scatterplot of the total ERCC molecule-counts and total gene molecule-counts.
The colors represent the three individuals (NA19098 is in red, NA19101 in green, and NA19239 in blue).
Data from different C1 replicates is plotted in different shapes.
(C and D) Violin plots of the reads to molecule conversion efficiency (total molecule-counts divided by total read-counts per single cells) by C1 replicate.
The endogenous genes and the ERCC spike-ins are shown separately in (C) and (D), respectively.
There is significant difference across individuals of both endogenous genes (*P* < 0.001) and ERCC spike-ins (*P* < 0.05).
The differences across C1 replicates per individual of endogenous genes and ERCC spike-ins were also evaluated (both *P* < 0.01).

**Figure `r fig_main_normalization`. Normalization and removal of technical variability.**
Principal component (PC) 1 versus PC2 of the (A) raw molecule counts, (B) log~2~ counts per million (cpm), (C) Poisson transformed expression levels (accounting for technical variability modeled by the ERCC spike-ins), and (D) batch-corrected expression levels.
The colors represent the three individuals (NA19098 in red, NA19101 in green, and NA19239 in blue).
Data from different C1 replicates is plotted in different shapes.

**Figure `r fig_main_noise`. Cell-to-cell variation in gene expression.**
Adjusted CV plotted against average molecule counts across all cells in (A) and across only the cells in which the gene is expressed (B), including data from all three individuals.
Each dot represents a gene, and the color indicates the corresponding gene-specific dropout rate (the proportion of cells in which the gene is undetected).
(C and D) Venn diagrams showing the overlaps of top 1000 genes across individuals based on mean expression level in (C) and based on adjusted CV values in (D), considering only the cells in which the gene is expressed.
(E and F) Similarly, Venn diagrams showing the overlaps of top 1000 genes across individuals based on mean expression level in (E) and based on adjusted CV values in (F), across all cells.
