---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Methods

### Ethics statement

The YRI cell lines were purchased from CCR.
The original samples were collected by the HapMap project between 2001-2005.
All of the samples were collected with extensive community engagement, including discussions with members of the donor communities about the ethical and social implications of human genetic variation research.
Donors gave broad consent to future uses of the samples, including their use for extensive genotyping and sequencing, gene expression and proteomics studies, and all other types of genetic variation research, with the data publicly released.

### Cell culture of iPSCs

Undifferentiated feeder-free iPSCs reprogrammed from LCLs of Yoruba individuals in Ibadan, Nigeria (abbreviation: YRI) [@HapMapConsortium2005] were grown in E8 medium (Life Technologies) [@Chen2011] on Matrigel-coated tissue culture plates with daily media feeding at 37 °C with 5% (vol/vol) CO2.
For standard maintenance, cells were split every 3-4 days using cell release solution (0.5 mM EDTA and NaCl in PBS) at the confluence of roughly 80%.
For the single cell suspension, iPSCs were individualized by Accutase Cell Detachment Solution (BD) for 5-7 minutes at 37 °C and washed twice with E8 media immediately before each experiment.
Cell viability and cell counts were then measured by the Automated Cell Counter (Bio-Rad) to generate resuspension densities of 2.5 X 105 cells/mL in E8 medium for C1 cell capture.

### Single cell capture and library preparation

Single cell loading and capture were performed following the Fluidigm protocol (PN 100-7168).
Briefly, 30 $\mu$l of C1 Suspension Reagent was added to a 70-$\mu$l aliquot of ~17,500 cells.
Five $\mu$l of this cell mix were loaded onto 10-17 $\mu$m C1 Single-Cell Auto Prep IFC microfluidic chip (Fluidigm), and the chip was then processed on a C1 instrument using the cell-loading script according to the manufacturer's instructions.
Using the standard staining script, the iPSCs were stained with StainAlive TRA-1-60 Antibody (Stemgent, PN 09-0068).
The capture efficiency and TRA-1-60 staining were then inspected using the EVOS FL Cell Imaging System (Thermo Fisher) (Supplementary Table `r table_collection`).

Immediately after imaging, reverse transcription and cDNA amplification were performed in the C1 system using the SMARTer PCR cDNA Synthesis kit (Clontech) and the Advantage 2 PCR kit (Clontech) according to the instructions in the Fluidigm user manual with minor changes to incorporate UMI labeling [@Islam2014].
Specifically, the reverse transcription primer and the 1:50,000 Ambion® ERCC Spike-In Mix1 (Life Technologies) were added to the lysis buffer, and the template-switching RNA oligos which contain the UMI (5-bp random sequence) were included in the reverse transcription mix [@Islam2011; @Islam2012; @Islam2014].
When the run finished, full-length, amplified, single-cell cDNA libraries were harvested in a total of approximately 13 $\mu$l C1 Harvesting Reagent and quantified using the DNA High Sensitivity LabChip (Caliper).
The average yield of samples per C1 plate ranged from 1.26-1.88 ng per microliter (Supplementary Table `r table_collection`).
A bulk sample, a 40 $\mu$l aliquot of ~10,000 cells, was collected in parallel with each C1 chip using the same reaction mixes following the C1 protocol (PN 100-7168, Appendix A).

For sequencing library preparation, tagmentation and isolation of 5' fragments were performed according to the UMI protocol [@Islam2014].
Instead of using commercially available Tn5 transposase, Tn5 protein stock was freshly purified in house using the IMPACT system (pTXB1, NEB) following the protocol previously described [@Picelli2014].
The activity of Tn5 was tested and shown to be comparable with the EZ-Tn5-Transposase (Epicentre).
Importantly, all the libraries in this study were generated using the same batch of Tn5 protein purification.
For each of the bulk samples, two libraries were generated using two different indices in order to get sufficient material for sequencing.
All 18 bulk libraries were then pooled and labeled as the "bulk" for sequencing.

### Illumina high-throughput sequencing

The scRNA-seq libraries generated from the 96 single cell samples of each C1 chip were pooled and then sequenced in three lanes on an Illumina HiSeq 2500 instrument using the PCR primer (C1-P1-PCR-2: Bio-GAATGATACGGCGACCACCGAT) as the read 1 primer and the Tn5 adapter (C1-Tn5-U: PHO-CTGTCTCTTATACACATCTGACGC) as the index read primer following the UMI protocol [@Islam2014].

The master mixes, one mix with all the bulk samples and nine mixes corresponding to the three replicates for the three individuals, were sequenced across four flowcells using a design aimed to minimize the introduction of technical batch effects (Supplementary Table `r table_collection`).
Single-end 100 bp reads were generated along with 8-bp index reads corresponding to the cell-specific barcodes.
We did not observe any obvious technical effects due to sequencing lane or flow cell that confounded the inter-individual and inter-replicate comparisons.

### Read mapping

To assess read quality, we ran FastQC ([http://www.bioinformatics.babraham.ac.uk/projects/fastqc][fastqc]) and observed a decrease in base quality at the 3' end of the reads.
Thus we removed low quality bases from the 3' end using sickle with default settings [@Joshi2011].
To handle the UMI sequences at the 5' end of each read, we used umitools [@umitools] to find all reads with a UMI of the pattern NNNNNGGG (reads without UMIs were discarded).
We then mapped reads to human genome hg19 (only including chromosomes 1-22, X, and Y, plus the ERCC sequences) with Subjunc [@Liao2013], discarding non-uniquely mapped reads (option -u).
To obtain gene-level counts, we assigned reads to protein-coding genes (Ensembl GRCh37 release 82) and the ERCC spike-in genes using featureCounts [@Liao2014].
Because the UMI protocol maintains strand information, we required that reads map to a gene in the correct orientation (featureCounts flag -s 1).

In addition to read counts, we utilized the UMI information to obtain molecule counts for the single cell samples.
We did not count molecules for the bulk samples because this would violate the assumptions of the UMI protocol, as bulk samples contain far too many unique molecules for the 1,024 UMIs to properly tag them all.
First, we combined all reads for a given single cell using samtools [@Li2009].
Next, we converted read counts to molecule counts using UMI-tools [@Smith2016].
UMI-tools counts the number of UMIs at each read start position.
Furthermore, it accounts for sequencing errors in the UMIs introduced during the PCR amplification or sequencing steps using a "directional adjacency" method.
Briefly, all UMIs at a given read start position are connected in a network using an edit distance of one base pair.
However, edges between nodes (the UMIs) are only formed if the nodes have less than a 2x difference in reads.
The node with the highest number of reads is counted as a unique molecule, and then it and all connected nodes are removed from the network.
This is repeated until all nodes have been counted or removed.

[fastqc]: http://www.bioinformatics.babraham.ac.uk/projects/fastqc

### Filtering cells and genes

We performed multiple quality control analyses to detect and remove data from low quality cells.
In an initial analysis investigating the percentage of reads mapping to the ERCC spike-in controls, we observed that replicate 2 of individual NA19098 was a clear outlier (Supplementary Fig. `r fig_supp_qc`).
It appeared that too much ERCC spike-in mix was added to this batch, which violated the assumption that the same amount of ERCC molecules was added to each cell.
Thus, we removed this batch from all of our analyses.

Next, we kept data from high quality single cells that passed the following criteria:

*  Only one cell observed per well
*  At least 1,556,255 mapped reads
*  Less than 36.4% unmapped reads
*  Less than 3.2% ERCC reads
*  More than 6,788 genes with at least one read

We chose the above criteria based on the distribution of these metrics in the empty wells (the cutoff is the 95th percentile, Supplementary Fig. `r fig_supp_qc`).
In addition, we observed that some wells classified as containing only one cell were clustered with multi-cell wells when plotting 1) the number of gene molecules versus the concentration of the samples, and 2) the read to molecule conversion efficiency (total molecule number divided by total read number) of endogenous genes versus that of ERCC.
We therefore established filtering criteria for these misidentified single-cell wells using linear discriminant analysis (LDA).
Specifically, LDA was performed to classify wells into empty, one-cell, and two-cell using the discriminant functions of 1) sample concentration and the number of gene molecules, and 2) endogenous and ERCC gene read to molecule conversion efficiency (Supplementary Fig. `r fig_supp_lda`).
After filtering, we maintained `r nrow(anno_filter)` high quality single cells
(NA19098: `r sum(anno_filter$individual == "NA19098")`,
 NA19101: `r sum(anno_filter$individual == "NA19101")`,
 NA19239: `r sum(anno_filter$individual == "NA19239")`).

The quality control analyses were performed using all protein-coding genes (Ensembl GRCh37 release 82) with at least one observed read.
Using the high quality single cells, we further removed genes with low expression levels for downstream analyses.
We removed all genes with a mean log~2~ cpm less than 2, which did not affect the relative differences in the proportion of genes detected across batches (Supplementary Fig. `r fig_supp_proportion`).
We also removed genes with molecule counts larger than 1,024 for the correction of collision probability.
In the end we kept
`r format(sum(grepl("ENSG", rownames(molecules_filter))), big.mark = ",")`
endogenous genes and
`r sum(grepl("ERCC", rownames(molecules_filter)))`
ERCC spike-in genes.

### Calculate the input molecule quantities of ERCC spiked-ins

According to the information provided by Fluidigm, each of the 96 capture chamber received 13.5 nl of lysis buffer, which contain 1:50,000 Ambion® ERCC Spike-In Mix1 (Life Technologies) in our setup.
Therefore, our estimation of the total spiked-in molecule number was 16,831 per sample.
Since the relative concentrations of the ERCC genes were provided by the manufacturer, we were able to calculate the molecule number of each ERCC gene added to each sample.
We observed that the levels of ERCC spike-ins strongly correlated with the input quantities (r = 0.9914, Fig. `r fig_main_qc`G).
The capture efficiency, defined as the fraction of total input molecules being successfully detected in each high quality cell, had an average of 6.1%.

### Subsampling

We simulated different sequencing depths by randomly subsampling reads and processing the subsampled data through the same pipeline described above to obtain the number of molecules per gene for each single cell.
To assess the impact of sequencing depth and number of single cells, we calculated the following three statistics:

1.  The Pearson correlation of the gene expression level estimates from the single cells compared to the bulk samples.
For the single cells, we summed the gene counts across all the samples and then calculated the log~2~ cpm of this pseudo-bulk.
For the bulk samples, we calculated the log~2~ cpm separately for each of the three replicates and then calculated the mean per gene.
2.  The number of genes detected with at least one molecule in at least one cell.
3.  The Pearson correlation of the cell-to-cell gene expression variance estimates from the subsampled single cells compared to the variance estimates using the full single cell data set.

Each data point in Fig. `r fig_main_subsample` represents the mean +/- the standard error of the mean (SEM) of 10 random subsamples of cells.
We split the genes by expression level into two groups (6,097 genes each) to highlight that most of the improvement with increased sequencing depth and number of cells was driven by the estimates of the lower half of expressed genes.
The data shown is for individual NA19239, but the results were consistent for individuals NA19098 and NA19101.
Only high quality single cells (Supplementary Table `r table_qualitycell`) were included in this analysis.

### A framework for testing individual and batch effects

Individual effect and batch effect between the single cell samples were evaluated in a series of analyses that examine the potential sources of technical variation on gene expression measurements.
These analyses took into consideration that in our study design, sources of variation between single cell samples naturally fall into a hierarchy of individuals and C1 batches.
In these sample-level analyses, the variation introduced at both the individual-level and the batch-level was modeled in a nested framework that allows random noise between C1 batches within individuals.
Specifically, for each cell sample in individual $i$, replicate $j$ and well $k$, we used $y_{ijk}$ to denote some sample measurement (e.g. total molecule-counts) and fit a linear mixed model with the fixed effect of individual $\alpha_i$ and the random effect of batch $b_{ij}$:

$$y_{ijk} = \alpha_{i} + b_{ij} + \epsilon_{ijk} \,\,\,\,(1)$$

where the random effect $b_{ij}$ of batch follows a normal distribution with mean zero and variance $\sigma^2_{b}$, and $\epsilon_{ijk}$ describes residual variation in the sample measurement.
To test the statistical significance of individual effect (i.e., null hypothesis $\alpha_1 = \alpha_2 = \alpha_3$), we performed a likelihood ratio test (LRT) to compare the above full model and the reduced model that excludes $\alpha_i$.
To test if there was a batch effect (i.e., null hypothesis  $\sigma^2_b = 0$), we performed an F-test to compare the variance that is explained by the above full model and the variance due to the reduced model that excludes $b_{ij}$.

The nested framework was applied to test the individual and batch effects between samples in the following cases.
The data includes samples after quality control and filtering.

1. Total molecule count (on the log~2~ scale) was modeled as a function of individual effect and batch effect, separately for the ERCC spike-ins and for the endogenous genes.

2. Read-to-molecule conversion efficiency was modeled as a function of individual effect and batch effect, separately for the ERCC spike-ins and for the endogenous genes.

### Estimating variance components for per-gene expression levels

To assess the relative contributions of individual and technical variation, we analyzed per-gene expression profiles and computed variance component estimates for the effects of individual and C1 batch (Supplementary Fig. `r fig_supp_variance`).
The goal here was to quantify the proportion of cell-to-cell variance due to individual (biological) effect and to C1 batch (technical) at the per-gene level.
Note that the goal here was different from that of the previous section, where we simply tested for the existence of individual and batch effects at the sample level by rejecting the null hypothesis of no such effects.
In contrast, here we fit a linear mixed model per gene where the dependent variable was the gene expression level (log~2~ counts per million) and the independent variables were individual and batch, both modeled as random effects.

The variance parameters of individual effect and batch effect were estimated using a maximum penalized likelihood approach [@Chung2013], which can effectively avoid the common issue of zero variance estimates due to small sample sizes (there were three individuals and eight batches).
We used the blmer function in the R package blme and set the penalty function to be the logarithm of a gamma density with shape parameter = 2 and rate parameter tending to zero.

The estimated variance components were used to compute the sum of squared deviations for individual and batch effects.
The proportion of variance due to each effect is equal to the relative contribution of the sum of squared deviations for each effect compared to the total sum of squared deviations per gene.
Finally, we compared the estimated proportions of variance due to the individual effect and the batch effect, across genes, using a non-parametric one-way analysis of variance (Kruskal-Wallis rank sum test).

### Normalization

We transformed the single cell molecule counts in multiple steps (Fig. `r fig_main_normalization`).
First, we corrected for the collision probability using a method similar to that developed by Grün et al. [@Grun2014].
Essentially we corrected for the fact that we did not observe all the molecules originally in the cell.
The main difference between our approach and that of Grün et al. [@Grun2014] was that we applied the correction at the level of gene counts and not individual molecule counts.
Second, we standardized the molecule counts to log~2~ counts per million (cpm).
This standardization was performed using only the endogenous gene molecules and not the ERCC molecules.
Third, we corrected for cell-to-cell technical noise using the ERCC spike-in controls.
For each single cell, we fit a Poisson generalized linear model (GLM) with the log~2~ expected ERCC molecule counts as the independent variable, and the observed ERCC molecule counts as the dependent variable, using the standard log link function.
Next we used the slope and intercept of the Poisson GLM regression line to transform the log~2~ cpm for the endogenous genes in that cell.
This is analogous to the standard curves used for qPCR measurements, but taking into account that lower concentration ERCC genes will have higher variance from Poisson sampling.
Fourth, we removed technical noise between the eight batches (three replicates each for NA19101 and NA19239 and two replicates for NA19098).
We fit a linear mixed model with a fixed effect for individual and a random effect for the eight batches and removed the variation captured by the random effect (see the next section for a detailed explanation).

For the bulk samples, we used read counts even though the reads contained UMIs.
Because these samples contained RNA molecules from ~10,000 cells, we could not assume that the 1,024 UMIs were sufficient for tagging such a large number of molecules.
We standardized the read counts to log~2~ cpm.

### Removal of technical batch effects

Our last normalization step adjusted the transformed log~2~ gene expression levels for cell-to-cell correlation within each C1 plate.
The algorithm mimics a method that was initially developed for adjusting within-replicate correlation in microarray data [@Smyth2005]. We assumed that for each gene $g$, cells that belong to the same batch $j$ are correlated, for batches $j = 1, \dots, 8$.
We also assume that the cell-to-cell gene expression variation due to C1 batch effect is independent of biological variation between individuals.

We fit a linear mixed model for each gene $g$ that includes a fixed effect of individual and a random effect for within-batch variation attributed to cell-to-cell correlation in each C1 plate:

$$ y_{g,ijk} = \mu_{g} + \alpha_{g,i} + b_{g,ij} + \epsilon_{g,ijk}, \,\,\,\,(2)$$

where $y_{g,ijk}$ denotes log~2~ counts-per-million (cpm) of gene $g$ in individual $i$, replicate $j$, and cell $k$; $i = NA19098, NA19101, NA19239$, $j = 1, \dots, n_i$ with $n_i$ the number of replicates in individual $i$, $k = 1, \dots, n_{ij}$ with $n_{ij}$ the number of cells in individual $i$ replicate $j$.
$\mu_g$ denotes the mean gene expression level across cells, $\alpha_{g,i}$ quantifies the individual effect on mean gene expression, $b_{g,ij}$ models the replicate effect on mean expression level (assumed to be stochastic, independent, and identically distributed with mean 0 and variance $\sigma^2_{g,b}$).
Finally, $\epsilon_{g,ijk}$ describes the residual variation in gene expression.

Batch-corrected expression levels were computed as

$$ \widehat{y}_{g,ijk} = y_{g,ijk} - \widehat{b}_{g,ij}, \,\,\,\,(3)$$

where $\widehat{b}_{g,ij}$ are the least-squares estimates.
The computations in this step were done with the gls.series function of the limma package [@limma].
We note that the batch correction method described here relies on the assumption that the technical effect due to batch is identical across individuals (no interaction).
This assumption may be violated, yet due to the small number of individuals in our study, we are unable to explicitly test for a first order interaction between batch and individual.
In the future, the issue of confounding effect between individual and batch may be more effectively addressed in the design of the study, instead of increasing the sample size.
Indeed, in the main text we describe a study design that can effectively address the issue of confounding effect by processing samples from multiple individuals in a single C1 plate.

### Measurement of gene expression noise

While examining gene expression noise (using the coefficient of variation or CV) as a function of mean RNA abundance across C1 replicates, we found that the CV of molecule counts among endogenous genes and ERCC spike-in controls suggested similar expression variability patterns.
Both endogenous and ERCC spike-in control CV patterns approximately followed an over-dispersed Poisson distribution (Supplementary Fig. `r fig_supp_CV`), which is consistent with previous studies [@Islam2014; @Brennecke2013].
We computed a measure of gene expression noise that is independent of RNA abundance across individuals [@Kolodziejczyk2015; @Newman2006].
First, squared coefficients of variation (CVs) for each gene were computed for each individual and also across individuals, using the batch-corrected molecule data.
Then we computed the distance of individual-specific CVs to the rolling median of global CVs among genes that have similar RNA abundance levels.
These transformed individual CV values were used as our measure of gene expression noise.
Specifically, we computed the adjusted CV values as follows:

1. Compute squared CVs of molecule counts in each individual and across individuals.

2. Order genes by the global average molecule counts.

3. Starting from the genes with the lowest global average gene expression level, for every sliding window of 50 genes, subtract log~10~ median squared CVs from log~10~ squared CVs of each cell line, and set 25 overlapping genes between windows.
The computation was performed with the rollapply function of the R zoo package [@Zeileis2005].
After this transformation step, CV no longer had a polynomial relationship with mean gene molecule count (Supplementary Fig. `r fig_supp_CV`).

### Identification of genes associated with inter-individual differences in regulatory noise

To identify differential noise genes across individuals, we computed median absolute deviation (MAD) - a robust and distribution-free dissimilarity measure for gene $g$:

$$ MAD_{g} = Median_{i= 1,2,3} \left| \text{adjCV}_{g,i} -  Median_{i= 1,2,3} ({\text{adjCV}}_{g,i}) \right|. \,\,\,\,(4)$$

Large values of $MAD_{g}$ suggest a large deviation from the median of the adjusted CV values.
We identified genes with significant inter-individual differences using a permutation-based approach.
Specifically, for each gene, we computed empirical *P*-values based on 300,000 permutations.
In each permutation, the sample of origin labels were shuffled between cells.
Because the number of permutations in our analysis was smaller than the maximum possible number of permutations, we computed the empirical *P*-values as $\frac{b + 1}{m + 1}$, where *b* is the number of permuted MAD values greater than the observed MAD value, and *m* is the number of permutations.
Adding 1 to *b* avoided an empirical *P*-value of zero [@Phipson2010].

### Gene enrichment analysis

We used ConsensusPATHDB [@Kamburov2011] to identify GO terms that are over-represented for genes whose
variation in single cell expression levels were significantly difference between individuals.

### Individual assignment based on scRNA-seq reads

We were able to successfully determine the correct identity of each single cell sample by examining the SNPs present in their RNA sequencing reads.
Specifically, we used the method verifyBamID ([https://github.com/statgen/verifyBamID][verifyBamID]) developed by Jun et al., 2012 [@Jun2012], which detects sample contamination and/or mislabeling by comparing the polymorphisms observed in the sequencing reads for a sample to the genotypes of all individuals in a study.
For our test, we included the genotypes for all 120 Yoruba individuals that are included in the International HapMap Project [@HapMapConsortium2005].
The genotypes included the HapMap SNPs with the 1000 Genomes Project SNPs [@OneKGConsortium2012] imputed, as previously described [@McVicker2013].
We subset to include only the 528,289 SNPs that overlap Ensembl protein-coding genes.
verifyBamID used only 311,848 SNPs which passed its default thresholds (greater than 1% minor allele frequency and greater than 50% call rate).
Using the option --best to return the best matching individual, we obtained 100% accuracy identifying the single cells of all three individuals (Supplementary Fig. `r fig_supp_design`).

[verifyBamID]: https://github.com/statgen/verifyBamID

### Data and code availability

The data have been deposited in NCBI's Gene Expression Omnibus [@Edgar2002] and are accessible through GEO Series accession number GSE77288 ([http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE77288][geo]).
The code and processed data are available at [https://github.com/jdblischak/singleCellSeq][repo].
The results of our analyses are viewable at [https://jdblischak.github.io/singleCellSeq/analysis][site].

[geo]: http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE77288
[repo]: https://github.com/jdblischak/singleCellSeq
[site]: https://jdblischak.github.io/singleCellSeq/analysis

## Acknowledgments
We thank Xin He and members of the Pritchard, Gilad, and Stephens laboratories for valuable discussions during the preparation of this manuscript.
This work was funded by NIH grant HL092206 to YG and HHMI funds to JKP.
PYT is supported by NIH T32HL007381.
JDB was supported by NIH T32GM007197.
The content is solely the responsibility of the authors and does not necessarily represent the official views of the National Institutes of Health.

## Author Contributions

YG and JKP conceived of the study,
designed the experiments,
and supervised the project.
PT and JEB performed the experiments.
PT, JDB, CH, and DAK analyzed the results.
PT, JDB, CH, and YG wrote the original draft.
All authors reviewed the final manuscript.

## Competing financial interests

The authors declare no competing financial interests.
