---
bibliography: refs.bib
---

```{r chunk-options, include=FALSE}
source("chunk-options-paper.R")
```

## Results
### Study design and quality control
We collected single cell RNA-seq (scRNA-seq) data from three YRI iPSC lines using the Fluidigm C1 microfluidic system followed by sequencing.
We added ERCC spike-in controls to each sample, and used 5-bp random sequence UMIs to allow for the direct quantification of mRNA molecule numbers.
For each of the YRI lines, we performed three independent C1 collections;
each replicate was accompanied by processing of a matching bulk sample using the same reagents.
This study design (Fig. `r fig_main_qc`A and Supplementary Table `r table_collection`) allows us to estimate error and variability associated with the technical processing of the samples, independently from the biological variation across single cells of different individuals.
We were also able to estimate how well scRNA-seq data can recapitulate the RNA-seq results from population bulk samples.

In what follows, we describe data as originating from different samples when we refer to data from distinct wells of each C1 collection.
Generally, each sample corresponds to a single cell.
In turn, we describe data as originating from different replicates when we refer to all samples from a given C1 collection, and from different individuals when we refer to data from all samples and replicates of a given genetically distinct iPSC line.

```{r calculate-seq-depth}
library("dplyr")
total_counts <- read.table("../data/total-counts.txt", header = TRUE,
                           sep = "\t", stringsAsFactors = FALSE)
depth_stats <- total_counts %>%
  filter(stage == "raw", well != "bulk") %>%
  group_by(individual, replicate, well) %>%
  summarize(counts_per_sample = sum(counts)) %>%
  ungroup() %>%
  summarize(mean = mean(counts_per_sample), sd = sd(counts_per_sample),
            min = min(counts_per_sample), max = max(counts_per_sample))
```

We obtained an average of `r round(depth_stats$mean / 10^6, digits = 1)` +/-
`r round(depth_stats$sd / 10^6, digits = 1)` million sequencing reads per sample
(range `r round(depth_stats$min / 10^6, digits = 1)`-`r round(depth_stats$max / 10^6, digits = 1)` million reads).
We processed the sequencing reads using a standard alignment approach (see Methods) and performed multiple quality control analyses.
As a first step, we estimated the proportion of ERCC spike-in reads from each sample.
We found that, across samples, sequencing reads from practically all samples of the second replicate of individual NA19098 included unusually high ERCC content compared to all other samples and replicates (Supplementary Fig. `r fig_supp_qc`).
We concluded that a pipetting error led to excess ERCC content in this replicate and we excluded the data from all samples of this replicate in subsequent analyses.
With the exception of the excluded samples, data from all other replicates seem to have similar global properties (using general metrics; Fig. `r fig_main_qc`C-E and Supplementary Fig. `r fig_supp_qc`).

We next examined the assumption that data from each sample correspond to data from a single cell.
After the cell sorting was complete, but before the processing of the samples, we performed visual inspection of the C1 microfluidic plates.
Based on that visual inspection, we flagged 21 samples that did not contain any cell, and 54 samples that contained more than one cell (across all batches).
Visual inspection of the C1 microfluidic plate is an important quality control step, but it is not infallible.
We therefore filtered data from the remaining samples based on the number of total mapped reads, the percentage of unmapped reads, the percentage of ERCC spike-in reads, and the number of genes detected (Fig. `r fig_main_qc`B-E).
We chose data-driven inclusion cutoffs for each metric, based on the 95th percentile of the respective distributions for the 21 libraries that were amplified from samples that did not include a cell based on visual inspection (Supplementary Fig. `r fig_supp_qc`).
Using this approach, we identified and removed data from 15 additional samples that were classified as originating from a single cell based on visual inspection, but whose data were more consistent with a multiple-cell origin based on the number of total molecules, the concentration of cDNA amplicons, and the read-to-molecule conversion efficiency (defined as the number of total molecules divided by the number of total reads; Supplementary Fig. `r fig_supp_lda`).
At the conclusion of these quality control analyses and exclusion steps, we retained data from `r nrow(anno_filter)` high quality samples, which correspond, with reasonable confidence, to `r nrow(anno_filter)` single cells, across eight replicates from three individuals (Supplementary Table `r table_qualitycell`).

Our final quality check focused on the different properties of sequencing read and molecule count data.
We considered data from the 564 high quality samples and compared gene specific counts of sequencing read and molecules.
We found that while gene-specific reads and molecule counts are exceptionally highly correlated when we considered the ERCC spike-in data (r = 0.99; Fig. `r fig_main_qc`F), these counts are somewhat less correlated when data from the endogenous genes are considered (r = 0.92).
Moreover, the gene-specific read and molecule counts correlation is noticeably lower for genes that are expressed at lower levels (Fig. `r fig_main_qc`F).
These observations concur with previous studies [@Islam2014; @Grun2014] as they underscore the importance of using UMIs in single cell gene expression studies.

```{r}
library("dplyr")
d <- read.table("../data/subsampling-results.txt",
                header = TRUE, sep = "\t", stringsAsFactors = FALSE)
d_grouped <- d %>%
  group_by(type, depth, gene_subset, subsampled_cells,
           individual, potential_cells, available_cells,
           lower_q, upper_q, available_ensg, used_ensg,
           available_ercc, used_ercc) %>%
  summarize(mean_detected = mean(detected_ensg),
            sem_detected = sd(detected_ensg) / sqrt(length(detected_ensg)),
            mean_bulk = mean(pearson_ensg),
            sem_bulk = sd(pearson_ensg) / sqrt(length(pearson_ensg)),
            mean_var = mean(var_pearson),
            sem_var = sd(var_pearson) / sqrt(length(var_pearson)))
d_filter <- d_grouped %>% filter(individual == "NA19239",
                                type == "molecules",
                                gene_subset == "lower",
                                depth == 1500000,
                                subsampled_cells == 75)

```

We proceeded by investigating the effect of sequencing depth and the number of single cells collected on multiple properties of the data.
To this end, we repeatedly subsampled single cells and sequencing reads to assess the correlation of the single cell gene expression estimates to the bulk samples, the number of genes detected, and the correlation of the cell-to-cell gene expression variance estimates between the reduced subsampled data and the full single cell gene expression data set (Fig. `r fig_main_subsample`).
We observed quickly diminishing improvement in all three properties with increasing sequencing depth and the number of sampled cells, especially for highly expressed genes.
For example, a per cell sequencing depth of 1.5 million reads (which corresponds to ~50,000 molecules) from each of 75 single cells was sufficient for effectively quantifying even the lower 50% of expressed genes.
To be precise, at this level of subsampling for individual NA19239, we were able to detect a mean of `r d_filter$mean_detected ` genes out of `r d_filter$used_ensg` genes expressed in the bulk samples (the bottom 50%; Fig. `r fig_main_subsample`B); the estimated single cell expression levels of these genes (summed across all cells) correlated with the bulk sample gene expression levels with a mean Pearson coefficient of  `r round(d_filter$mean_bulk, digits = 2)` (Fig. `r fig_main_subsample`A), and the estimated cell-to-cell variation in gene expression levels was correlated with the variation estimated from the full data set with a mean Pearson coefficient of `r round(d_filter$mean_var, digits = 2)` (Fig. `r fig_main_subsample`C).
We observed similar results when examining each batch separately (Supplementary Fig. `r fig_supp_subsample`).

### Batch effects associated with UMI-based single cell data
In the context of the C1 platform, typical study designs make use of a single C1 plate (batch/replicate) per biological condition.
In that case, it is impossible to distinguish between biological and technical effects associated with the independent capturing and sequencing of each C1 replicate.
We designed our study with multiple technical replicates per biological condition (individual) in order to directly and explicitly estimate the batch effect associated with independent C1 preparations (Fig. `r fig_main_qc`A).

As a first step in exploring batch effects, we examined the gene expression profiles across all single cells that passed our quality checks (as reported above) using raw molecule counts (without standardization).
Using principal component analysis (PCA) for visualization, we observed â€“ as expected - that the major source of variation in data from single cells is the individual origin of the sample.
Specifically, we found that the proportion of variance due to individual was larger (median: 8%) than variance due to C1 batch (median: 4%; Kruskal-Wallis test; *P* < 0.001, Supplementary Fig. `r fig_supp_variance`; see Methods for details of the variance component analysis).
Yet, variation due to C1 batch is also substantial - data from single cell samples within a batch are more correlated than that from single cells from the same individual but different batches (Kruskal-Wallis test; *P* < 0.001).

Could we account for the observed batch effects using the ERCC spike-in controls?
In theory, if the total ERCC molecule-counts are affected only by technical variability, the spike-ins could be used to correct for batch effects even in a study design that entirely confounds biological samples with C1 preparations.
To examine this, we first considered the relationship between total ERCC molecule-counts and total endogenous molecule-counts per sample.
If only technical variability affects ERCC molecule-counts, we expect the technical variation in the spike-ins (namely, variation between C1 batches) to be consistent, regardless of the individual assignment.
Indeed, we observed that total ERCC molecule-counts are significantly different between C1 batches (F-test; *P* < 0.001).
However, total ERCC molecule-counts are also quite different across individuals, when variation between batches is taken into account (LRT; *P* = 0.08; Fig. `r fig_main_batch`A).
This observation suggests that both technical and biological variation affect total ERCC molecule-counts.
In addition, while we observed a positive relationship between total ERCC molecule-counts and total endogenous molecule-counts per sample, this correlation pattern differed across C1 batches and across individuals (F-test; *P* < 0.001; Fig. `r fig_main_batch`B).

To more carefully examine the technical and biological variation of ERCC spike-in controls, we assessed the ERCC per-gene expression profile.
We observed that the ERCC gene expression data from samples of the same batch were more correlated than data from samples across batches (Kruskal-Wallis test; Chi-squared *P* < 0.001).
However, the proportion of variance explained by the individual was significantly larger than the variance due to C1 batch (median: 9% vs. 5%, Chi-squared test; *P* < 0.001, Supplementary Fig. `r fig_supp_variance`), lending further support to the notion that biological variation affects the ERCC spike in data.
Based on these analyses, we concluded that ERCC spike-in controls cannot be used to effectively account for the batch effect associated with independent C1 preparations.

We explored additional properties in the data that may be associated with a batch effect.
To do so, we focused on the read-to-molecule conversion rates, i.e. the rates at which sequencing reads are converted to molecule counts based on the UMI sequences.
We defined read-to-molecule conversion efficiency as the total molecule-counts divided by the total reads-counts in each sample, considering separately the reads/molecules that correspond to endogenous genes or ERCC spike-ins (Fig. `r fig_main_batch`C and `r fig_main_batch`D).
We observed a significant batch effect in the read-to-molecule conversion efficiency of both ERCC (F-test; *P* < 0.05) and endogenous genes (F-test; *P* < 0.001) across C1 replicates from the same individual.
Moreover, the difference in read-to-molecule conversion efficiency across the three individuals was significant not only for endogenous genes (LRT; *P* < 0.01, Fig. `r fig_main_batch`C) but also in the ERCC spike-ins (LRT; *P* < 0.01, Fig. `r fig_main_batch`D).

### Measuring regulatory noise in single-cell gene expression data
Our analysis indicated that there is a considerable batch effect in the single cell gene expression data collected from the C1 platform.
We thus sought an approach that would account for the batch effect and allow us to study biological properties of the single-cell molecule count-based estimates of gene expression levels, albeit in a small sample of just three individuals.
As a first step, we adjusted the raw molecule counts by using a Poisson approximation to account for the random use of identical UMI sequences in molecules from highly expressed genes (this was previously termed a correction for the UMI â€˜collision probabilityâ€™ [@Fu2011]).
We then excluded data from genes whose inferred molecule count exceeded 1,024 (the theoretical number of UMI sequences) â€“ this step resulted in the exclusion of data from 6 mitochondrial genes.

We next incorporated a standardization step by computing log transformed counts-per-million (cpm) to remove the effect of different sequencing depths, as is the common practice for the analysis of bulk RNA-seq data (Fig. `r fig_main_normalization`A and `r fig_main_normalization`B).
We used a Poisson generalized linear model to normalize the endogenous molecule log~2~ cpm values by the observed molecule counts of ERCC spike-ins across samples.
While we do not expect this step to account for the batch effect (as discussed above), we reasoned that the spike-ins allow us to account for a subset of technical differences between samples, for example, those that arise from differences in RNA concentration (Fig. `r fig_main_normalization`C).

Finally, to account for the technical batch effect, we modeled between-sample correlations in gene expression within C1 replicates (see Methods).
Our approach is similar in principle to limma, which was initially developed for adjusting within-replicate correlations in microarray data [@Smyth2005].
We assume that samples within each C1 replicate share a component of technical variation, which is independent of biological variation across individuals. We fit a linear mixed model for each gene, which includes a fixed effect for individual and a random effect for batch.
The batch effect is specific to each C1 replicate, and is independent of biological variation across individuals. We use this approach to estimate and remove the batch effect associated with different C1 preparations (Fig. `r fig_main_normalization`D).

Once we removed the unwanted technical variability, we focused on analyzing biological variation in gene expression between single cells.
Our goal was to identify inter-individual differences in the amount of variation in gene expression levels across single cells,
or in other words, to identify differences between individuals in the amount of regulatory noise [@Raser2005].
In this context, regulatory noise is generally defined as the coefficient of variation (CV) of the gene expression levels of single cells [@Fehrmann2013].
In the following, we used the standardized, normalized, batch-corrected molecule count gene expression data to estimate regulatory noise (Fig. `r fig_main_normalization`D).
To account for heteroscedasticity from Poisson sampling, we adjusted the CV values by the average gene-specific expression level across cells of the same individual.
The adjusted CV is robust both to differences in gene expression levels, as well as to the proportion of gene dropouts in single cells.

To investigate the effects of gene dropouts (the lack of molecule representation of an expressed gene [@Brennecke2013; @Shalek2013]) on our estimates of gene expression noise, we considered the association between the proportion of cells in which a given gene is undetected (namely, the gene-specific dropout rate), the average gene expression level, and estimates of gene expression noise.
Across all genes, the median gene-specific dropout was 22 percent.
We found significant individual differences (LRT; *P* < 10^-5^) in gene-specific dropout rates between individuals in more than 10% (1,214 of `r format(sum(grepl("ENSG", rownames(molecules_filter))), big.mark = ",")`) of expressed endogenous genes.
As expected, the expression levels, and the estimated variation in expression levels across cells, are both associated with gene-specific dropout rates (Supplementary Fig. `r fig_supp_dropout`).
However, importantly, adjusted CVs are not associated with dropout rates (Spearman's correlation = 0.04; Supplementary Fig. `r fig_supp_dropout`), indicating that adjusted CV measurements are not confounded by the dynamic range of single-cell gene expression levels.

We thus estimated mean expression levels and regulatory noise (using adjusted CV) for each gene, by either including (Fig. `r fig_main_noise`A) or excluding (Fig. `r fig_main_noise`B) samples in which the gene was not detected/expressed.
We first focused on general trends in the data.
We ranked genes in each individual by their mean expression level as well as by their estimated level of variation across single cells.
When we considered samples in which a gene was expressed, we found that 887 of the 1,000 most highly expressed genes in each individual are common to all three individuals (Fig. `r fig_main_noise`C).
In contrast, only 95 of the 1,000 most highly variable (noisy) genes in each individual were common to all three individuals (Fig. `r fig_main_noise`D).
We found similar results when we considered data from all single cells, regardless of whether the gene was detected as expressed (Fig. `r fig_main_noise`E and `r fig_main_noise`F). In particular, the set of 887 highly expressed genes was also detected as common to all three individuals, while 80% of the 95 most highly variable genes (76 genes) were detected as common to all three individuals when including samples in which the gene was not detected as expressed.

Next, we identified genes whose estimated regulatory noise (based on the adjusted CV) is significantly different between individuals.
For the purpose of this analysis, we only included data from cells in which the gene was detected as expressed.
Based on permutations (Supplementary Fig. `r fig_supp_permutation`), we classified the estimates of regulatory noise of 560 genes as significantly different across individuals (empirical *P* < .0001, Supplementary Fig. `r fig_supp_noisygene` for examples; Supplementary Table `r table_noisygene` for gene list).
These 560 genes are enriched for genes involved in protein translation, protein disassembly, and various biosynthetic processes (Supplementary Table `r table_GO`).
Interestingly, among the genes whose regulatory noise estimates differ between individuals, we found two pluripotency genes, *KLF4* and *DPPA2* (Supplementary Fig. `r fig_supp_plurigene`).
